cfgov/cfgov/urls.py

```diff
@@ -413,53 +415,98 @@
-        url(r'^(?i)ask-cfpb/api/autocomplete/$',
-            ask_autocomplete, name='ask-autocomplete-en'),
-        url(r'^(?P<language>es)/obtener-respuestas/api/autocomplete/$',
-            ask_autocomplete, name='ask-autocomplete-es'),
+
+# New Spanish Ask CFPB URLs
+# These are defined here so they can be include()ed by the feature flag below
+ask_cfpb_spanish_patterns = [
+    url(r'^c/(.+)/(?P<ask_id>\d+)/(.+)\.html$',
+        RedirectView.as_view(
+            url='/es/obtener-respuestas/slug-es-%(ask_id)s',
+            permanent=True)),
+    url(r'^([-\w]{1,244})-(es)-(\d{1,6})/?$',
+        view_answer,
+        name='ask-spanish-answer'),
+    url(r'^([-\w]{1,244})-(es)-(\d{1,6})/imprimir/?$',  # noqa: E501
+        print_answer,
+        name='ask-spanish-print-answer'),
+    url(r'^buscar/$',
+        ask_search,
+        kwargs={'language': 'es'},
+        name='ask-search-es'),
+    url(r'^buscar/(?P<as_json>json)/$',  # noqa: E501
+        ask_search,
+        kwargs={'language': 'es'},
+        name='ask-search-es-json'),
+    url(r'^api/autocomplete/$',
+        ask_autocomplete,
+        kwargs={'language': 'es'},
+        name='ask-autocomplete-es'),
+
+    # Answer /es/obtener-respuestas landing page with Wagtail.
+    # This is necessary because the pattern is shared with the old
+    # knowledgebase urls and we have to explicitly offer it as an alternative
+    # in the flagged_url() for /es/obtener-respuestas below.
+    # Once the feature flag is removed this will be unnecessary.
+    url(r'^$',
+        lambda req: ServeView.as_view()(req, req.path)),
+]
+
+# New English Ask CFPB URL redirects
+# These are defined here so they can be include()ed by the feature flag below.
+ask_cfpb_english_redirects = [
+    url(r'^$',
+        RedirectView.as_view(
+            url='/ask-cfpb/',
+            permanent=True)),
+    url(r'^(?P<ask_id>\d+)/(.*)$',
+        RedirectView.as_view(
+            url='/ask-cfpb/slug-en-%(ask_id)s',
+            permanent=True)),
+    url(r'^search/\?selected_facets=category_exact:(?P<category>[^&]+)',  # noqa: E501
+        RedirectView.as_view(
+            url='/ask-cfpb/category-%(category)s',
+            permanent=True)),
+]
+
+# Flagged includes for either old knowledgebase Ask CFPB URLs or new ask_cfpb
+# URLs. In this way the shared URLs askcfpb/ and es/obtener-respuestas/ will
+# either serve the new ask_cfpb or will fallback on the old knowledgebase.
+flagged_kb_patterns = [
+    flagged_url('WAGTAIL_ASK_CFPB',
+                r'^(?i)askcfpb/',
+                include(ask_cfpb_english_redirects),
+                fallback=include_if_app_enabled(
+                    'knowledgebase', 'knowledgebase.urls')),
+    flagged_url('WAGTAIL_ASK_CFPB',
+                r'^es/obtener-respuestas/',
+                include(ask_cfpb_spanish_patterns),
+                fallback=include_if_app_enabled(
+                    'knowledgebase', 'knowledgebase.babel_urls')),
+]
```

Comment:
@willbarton The landing and question page redirects are working perfectly for me, but not the category pages (`askcfpb/search/?selected_facets=category_exact:auto-loans`). Maybe that's a local issue, though?